src/search.rs

```diff
@@ -9,6 +10,32 @@ pub struct MatchRange {
     pub length: usize,
 }
 
+#[derive(Debug, Eq, PartialEq , Clone)]
+pub enum Filter<'a> {
+    String(&'a str),
+    Array(Vec<&'a str>)
+}
```

Comment:
```suggestion
#[derive(Debug, Eq, PartialEq , Clone)]
pub struct Filter<'a>(
    #[serde(with = "either::serde_untagged")]
    inner: Either<&'a str, Vec<&'a str>>,
);
```

Hey @VoidCupboard, someone already wrote your serializer; it would be nice if you could instead use [`Either`](https://docs.rs/either/latest/either/serde_untagged/index.html)